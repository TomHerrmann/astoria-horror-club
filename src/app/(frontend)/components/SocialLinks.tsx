import {
  AHC_DISCORD_INVITE_URL,
  AHC_INSTAGRAM_URL,
  DISCORD_ICON_SVG_URL,
  INSTAGRAM_ICON_SVG_URL,
} from '@/app/constants'
import Image from 'next/image'

export function SocialLinks() {
  return (
    <div className="flex py-9 px-5 w-full justify-between">
      <span className="flex items-center">
        Join Our Discord{' '}
        <a href={AHC_DISCORD_INVITE_URL} target="_blank">
          <Image
            height={24}
            width={24}
            alt="discord social media icon"
            src={DISCORD_ICON_SVG_URL}
          />
        </a>
      </span>
      <span className="flex items-center">
        Follow Us On Instagram{' '}
        <a href={AHC_INSTAGRAM_URL} target="_blank">
          <Image
            height={24}
            width={24}
            alt="instagram social media icon"
            src={INSTAGRAM_ICON_SVG_URL}
          />
        </a>
      </span>
    </div>
  )
}
