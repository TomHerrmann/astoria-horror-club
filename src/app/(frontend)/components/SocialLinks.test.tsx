import { describe, it, expect, vi } from 'vitest'
import { render, screen } from '@testing-library/react'
import SocialLinks from './SocialLinks'

vi.mock('next/image', () => ({
  // eslint-disable-next-line @next/next/no-img-element
  default: (props: any) => <img {...props} />,
}))

describe('SocialLinks', () => {
  it('matches snapshot', () => {
    const { container } = render(<SocialLinks />)
    expect(container).toMatchSnapshot()
  })

  it('renders both social links with correct text and icons', () => {
    render(<SocialLinks />)

    // Check Discord link
    const discordLink = screen.getByRole('link', { name: /discord social media icon/i })
    expect(discordLink).toBeDefined()
    expect(screen.getByText('Join Our Discord')).toBeDefined()
    const discordIcon = screen.getByAltText('discord social media icon')
    expect(discordIcon).toBeDefined()

    // Check Instagram link
    const instagramLink = screen.getByRole('link', { name: /instagram social media icon/i })
    expect(instagramLink).toBeDefined()
    expect(screen.getByText('Follow Us On Instagram')).toBeDefined()
    const instagramIcon = screen.getByAltText('instagram social media icon')
    expect(instagramIcon).toBeDefined()
  })
})
