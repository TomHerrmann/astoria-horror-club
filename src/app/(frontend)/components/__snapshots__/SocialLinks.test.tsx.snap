// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialLinks > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col flex-wrap justify-center items-center w-full justify-between"
  >
    <span
      class="flex items-center mb-2 mr-3"
    >
      Join Our Discord
       
      <a
        href="https://discord.gg/HubAHgKx"
        target="_blank"
      >
        <img
          alt="discord social media icon"
          height="24"
          src="https://fzuxxxhgqwm9izz9.public.blob.vercel-storage.com/Discord-Symbol-Blurple-lqt9tohtjIt6VHGUs7tj0wSPGeNXBD.svg"
          width="24"
        />
      </a>
    </span>
    <span
      class="flex items-center mb-2"
    >
      Follow Us On Instagram
       
      <a
        href="https://www.instagram.com/astoriahorrorclub/"
        target="_blank"
      >
        <img
          alt="instagram social media icon"
          height="24"
          src="https://fzuxxxhgqwm9izz9.public.blob.vercel-storage.com/Instagram_Glyph_Gradient-cytRhDzf7XqHFRst9K38AotJg5pZ4i.svg"
          width="24"
        />
      </a>
    </span>
  </div>
</div>
`;
